bcdaworker/worker/worker.go

```diff
@@ -100,7 +101,8 @@ func (w *worker) ProcessJob(ctx context.Context, job models.Job, jobArgs models.
 
 	// This is only run AFTER completion of all the collection
 	if err != nil {
-		err = w.r.UpdateJobStatus(ctx, job.ID, models.JobStatusFailed)
+		// only inProgress jobs should move to a failed status (i.e. don't move a cancelled job to failed)
+		err = w.r.UpdateJobStatusCheckStatus(ctx, job.ID, models.JobStatusInProgress, models.JobStatusFailed)
```

Comment:
Can we also update the call where we move from the "In Progress" to the "Completed" state to ensure that we do not move from cancelled to Completed?

Reply:
Can we also check for ErrJobNotUpdated? I think it shouldn't be an error scenario if the job status does not match up

if goerrors.Is(err, repository.ErrJobNotUpdated)

Reply:
tweaked this a bit - UpdateJobStatusCheckStatus was causing test failures because we currently allow for Failed to jobs to be marked as Completed, in addition to Pending jobs. So, I kept it as UpdateJobStatus but added a check for job.Status != Cancelled first, with a log indicating the Cancelled job was not updated to Completed

Reply:
Can we go ahead and remove/clean up those failing tests? I think the failed -> completed would also be an invalid case.

Or, since it's behavior that existed before, can we go ahead and make a ticket to fix this behavior?

Reply:
Sure, I'll write up a new ticket for it!